packages/cli/generators/model/index.js

```diff
@@ -0,0 +1,273 @@
+// Copyright IBM Corp. 2017,2018. All Rights Reserved.
+// Node module: @loopback/cli
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+'use strict';
+
+const ArtifactGenerator = require('../../lib/artifact-generator');
+const debug = require('../../lib/debug')('model-generator');
+const utils = require('../../lib/utils');
+const updateIndex = require('../../lib/update-index');
+const chalk = require('chalk');
+const path = require('path');
+const _ = require('lodash');
+
+/**
+ * Model Generator == CLI
+ *
+ * Prompts for a Model name, Model Base Class (currently defaults to 'Entity').
+ * Creates json file, a Base class for the model as well as a Class for a user
+ * to modify.
+ *
+ * Will prompt for properties to add to the Model till a blank property name is
+ * entered. Will also ask if a property is required, the default value for the
+ * property, if it's the ID (unless one has been selected), etc.
+ */
+module.exports = class ModelGenerator extends ArtifactGenerator {
+  constructor(args, opts) {
+    super(args, opts);
+  }
+
+  _setupGenerator() {
+    this.artifactInfo = {
+      type: 'model',
+      rootDir: 'src',
+    };
+
+    this.artifactInfo.outDir = path.resolve(
+      this.artifactInfo.rootDir,
+      'models',
+    );
+
+    // Model Property Types
+    this.typeChoices = [
+      'string',
+      'number',
+      'boolean',
+      'object',
+      'array',
+      'date',
+      'buffer',
+      'geopoint',
+      'any',
+    ];
+
+    this.artifactInfo.properties = {};
+
+    return super._setupGenerator();
+  }
+
+  checkLoopBackProject() {
+    return super.checkLoopBackProject();
+  }
+
+  promptArtifactName() {
+    return super.promptArtifactName();
+  }
+
+  promptBase() {
+    this.artifactInfo.className = utils.toClassName(this.artifactInfo.name);
+    const baseOptions = ['Entity'];
+    const prompts = [
+      {
+        name: 'base',
+        message: `Select the base class for ${chalk.yellow(
+          this.artifactInfo.className,
+        )}:`,
+        type: 'list',
+        default: baseOptions[0],
+        choices: baseOptions,
+        when: baseOptions.length > 1,
```

Comment:
I usually prefer to leave out code that will be used only later in the future because of [YAGNI](https://www.martinfowler.com/bliki/Yagni.html), but don't mind too much to have it here.

Reply:
Took it out -- it's simple enough logic that it can be added without pain when needed.